### PR TITLE
Bluetooth: controller: Fix ll_cig_remove not cleaning up

### PR DESCRIPTION
**Pull request by mtpr-ot**
When removing a CIG, the instance and associated CIS instances were not always released. This causes repeated Create_CIG/Remove_CIG operations to fail assertion with resource error.

Fixes issue #53301.

Signed-off-by: Morten Priess <mtpr@oticon.com>

### CODE REVIEW

**Review by MariuszSkamra — DISMISSED** [review]
I've ran bsim [Unicast Audio test](https://github.com/zephyrproject-rtos/zephyr/blob/8cd6c55d96ff0d5f599877df29e03c358930e216/tests/bluetooth/bsim_bt/bsim_test_audio/src/unicast_client_test.c) with `CONFIG_BT_CTLR_CENTRAL_ISO` enabled. The issue is not reproducible with this patch.

**Comment by cvinayak** [comment]
Is this needed in addition to a consolidated list of changes to support Broadcast and Connected ISO in this PR: https://github.com/zephyrproject-rtos/zephyr/pull/52955?

**Comment by mtpr-ot** [comment]
> Is this needed in addition to a consolidated list of changes to support Broadcast and Connected ISO in this PR: #52955?

In my opinion, the smaller the PRs we can submit to fix specific problems, the better traceability. Also, PR#52955 may take some time to get in, as we need to verify that it will still pass all CIS EBQ tests.

**Comment by cvinayak** [comment]
> > Is this needed in addition to a consolidated list of changes to support Broadcast and Connected ISO in this PR: #52955?
> 
> In my opinion, the smaller the PRs we can submit to fix specific problems, the better traceability. Also, PR #52955 may take some time to get in, as we need to verify that it will still pass all CIS EBQ tests.

#52955 is big as it is covered by upstream broadcast and unicast babblesim test.

I am not blocking this PR, just a comment that this PR change has no coverage upstream. Can a unit test be added?

**Comment by MariuszSkamra** [comment]
> I am not blocking this PR, just a comment that this PR change has no coverage upstream. Can a unit test be added?

Actually, there is a bsim test `tests/bluetooth/bsim_bt/bsim_test_audio/src/unicast_client_test.c` that can use it.
@mtpr-ot You could just add another change in this PR that enables `CONFIG_BT_CTLR_CENTRAL_ISO` in `bsim_test_audio`.

```
diff --git a/tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf b/tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf
index a7305b25cc..1c9830ea1e 100644
--- a/tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf
+++ b/tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf
@@ -166,3 +166,4 @@ CONFIG_BT_CTLR_ADV_ISO_STREAM_COUNT=1
 # Only 1 stream support by controller at this point
 CONFIG_BT_CTLR_SYNC_ISO_STREAM_MAX=1
 CONFIG_BT_CTLR_ISO_TX_BUFFERS=3
+CONFIG_BT_CTLR_CENTRAL_ISO=y
```

**Review by Thalley — CHANGES_REQUESTED** [review]
One minor comment, otherwise LGTM

File: tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf [hunk 1]
```diff
@@ -166,3 +166,4 @@ CONFIG_BT_CTLR_ADV_ISO_STREAM_COUNT=1
 # Only 1 stream support by controller at this point
 CONFIG_BT_CTLR_SYNC_ISO_STREAM_MAX=1
 CONFIG_BT_CTLR_ISO_TX_BUFFERS=3
+CONFIG_BT_CTLR_CENTRAL_ISO=y
```
Comment (Thalley) [inline tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf:169]: If we adding this, we could consider removing the guards from the audio tests, but that can be done in a separate PR

**Review by Thalley — DISMISSED** [review]

**Review by MariuszSkamra — DISMISSED** [review]

File: tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf (hunk 1 above)
Comment (Thalley) [reply tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf:169]: @mtpr-ot quite possible due to https://github.com/zephyrproject-rtos/zephyr/issues/53446

Last I tried, the data path wasn't working correctly if both BIS and CIS is enabled in the build

Comment (cvinayak) [reply tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf:169]: Enabling both CIS and BIS is supported in https://github.com/zephyrproject-rtos/zephyr/pull/52955, and the said babblesim test has been enabled for Connected ISO, passing in that PR.

(but, I have much more changes in local copy that now supports bidirectional encrypted connected ISO, streaming audio in our nRF5340 audio DK).

**Review by Thalley — APPROVED** [review]

**Review by thoh-ot — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Pull request merged**